src/windows/webview.c

```diff
@@ -477,47 +488,107 @@ static ICoreWebView2EnvironmentOptionsVtbl VTBL5 = {
 
 // Public
 
-static HMODULE webview_load_dll(void)
+static bool webview_dll_path_clientstate(wchar_t *pathw, bool as_user)
 {
-	HKEY key = NULL;
-	WCHAR *dll = NULL;
-	HMODULE lib = NULL;
+	bool ok = false;
+	DWORD flags = KEY_READ;
+
+	HKEY hkey = as_user ? HKEY_CURRENT_USER : HKEY_LOCAL_MACHINE;
+	if (!as_user)
+		flags |= KEY_WOW64_32KEY;
 
-	LSTATUS r = RegOpenKeyEx(HKEY_LOCAL_MACHINE,
-		L"Software\\Microsoft\\EdgeUpdate\\ClientState\\{F3017226-FE2A-4295-8BDF-00C3A9A7E4C5}",
-		0, KEY_WOW64_32KEY | KEY_READ, &key);
+	wchar_t reg_path[MAX_PATH] = {0};
+	_snwprintf_s(reg_path, MAX_PATH, _TRUNCATE, WEBVIEW_REG_PATH, L"ClientState");
+
+	HKEY key = NULL;
+	LSTATUS r = RegOpenKeyEx(hkey, reg_path, 0, flags, &key);
 	if (r != ERROR_SUCCESS)
 		goto except;
 
-	dll = MTY_Alloc(MAX_PATH, sizeof(WCHAR));
-	DWORD size = MAX_PATH * sizeof(WCHAR);
-
+	DWORD size = MAX_PATH * sizeof(wchar_t);
+	wchar_t dll[MAX_PATH] = {0};
 	r = RegQueryValueEx(key, L"EBWebView", 0, NULL, (BYTE *) dll, &size);
 	if (r != ERROR_SUCCESS)
 		goto except;
 
-	#if defined(_WIN64)
-		const WCHAR *path = L"\\EBWebView\\x64\\EmbeddedBrowserWebView.dll";
+	if (pathw)
+		_snwprintf_s(pathw, MTY_PATH_MAX, _TRUNCATE, L"%s\\%s", dll, WEBVIEW_DLL_PATH);
+
+	ok = true;
+
+	except:
+
+	if (key)
+		RegCloseKey(key);
+
+	return ok;
+}
+
+static bool webview_dll_path_client(wchar_t *pathw, bool as_user)
+{
+	bool ok = false;
+	DWORD flags = KEY_READ;
+
+	HKEY hkey = as_user ? HKEY_CURRENT_USER : HKEY_LOCAL_MACHINE;
+	if (!as_user)
+		flags |= KEY_WOW64_32KEY;
 
-	#else
-		const WCHAR *path = L"\\EBWebView\\x86\\EmbeddedBrowserWebView.dll";
-	#endif
+	wchar_t reg_path[MAX_PATH] = {0};
+	_snwprintf_s(reg_path, MAX_PATH, _TRUNCATE, WEBVIEW_REG_PATH, L"Clients");
 
-	if (wcscat_s(dll, MAX_PATH, path) != 0)
+	HKEY key = NULL;
+	LSTATUS r = RegOpenKeyEx(hkey, reg_path, 0, flags, &key);
+	if (r != ERROR_SUCCESS)
 		goto except;
 
-	lib = LoadLibrary(dll);
-	if (!lib)
+	DWORD size = MAX_PATH * sizeof(wchar_t);
+	wchar_t dll[MAX_PATH] = {0};
+	r = RegQueryValueEx(key, L"location", 0, NULL, (BYTE *) dll, &size);
+	if (r != ERROR_SUCCESS)
 		goto except;
 
-	except:
+	size = MAX_PATH * sizeof(wchar_t);
+	wchar_t version[MAX_PATH] = {0};
+	r = RegQueryValueEx(key, L"pv", 0, NULL, (BYTE *) version, &size);
+	if (r != ERROR_SUCCESS)
+		goto except;
+
+	if (pathw)
+		_snwprintf_s(pathw, MTY_PATH_MAX, _TRUNCATE, L"%s\\%s\\%s", dll, version, WEBVIEW_DLL_PATH);
 
-	MTY_Free(dll);
+	ok = true;
```

Comment:
```suggestion
	ok = PathFileExists(pathw);
```

Since if the system path is loaded from the registry, but doesn't exist, it will never try the user path.